Panic when TOC rendering.

A content for reproduce in the following.

```adoc
= sect0
:toc:

level 1 sections not exists.
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6bcdde]

goroutine 1 [running]:
github.com/bytesparadise/libasciidoc/pkg/renderer/html5.renderTableOfContents(0xc0001674a0, 0x739d80, 0x3074636573, 0x1, 0x1, 0xc0001c14b0)
        /go/src/github.com/bytesparadise/libasciidoc/pkg/renderer/html5/table_of_contents.go:56 +0x9e
github.com/bytesparadise/libasciidoc/pkg/renderer/html5.renderElement(0xc0001674a0, 0x731e20, 0xc215f0, 0x1, 0x1, 0xc05b40, 0x7f5f64039440, 0x0)
        /go/src/github.com/bytesparadise/libasciidoc/pkg/renderer/html5/renderer.go:79 +0x13ce
github.com/bytesparadise/libasciidoc/pkg/renderer/html5.renderElements(0xc0001674a0, 0xc0001d4120, 0x3, 0x3, 0x1, 0x5, 0x40, 0x0, 0x0)
        /go/src/github.com/bytesparadise/libasciidoc/pkg/renderer/html5/renderer.go:26 +0x124
github.com/bytesparadise/libasciidoc/pkg/renderer/html5.renderDocumentElements(0xc0001674a0, 0xc0001d40f0, 0xc0001d4120, 0x3, 0x3, 0xc0001d4150, 0xc215f0, 0x0, 0x0, 0xc0001d4180, ...)
        /go/src/github.com/bytesparadise/libasciidoc/pkg/renderer/html5/document.go:123 +0xe1
github.com/bytesparadise/libasciidoc/pkg/renderer/html5.renderDocument(0xc0001674a0, 0x7e5880, 0xc00009e008, 0x0, 0xc0001674a0, 0xc0001d41b0)
        /go/src/github.com/bytesparadise/libasciidoc/pkg/renderer/html5/document.go:96 +0x221
github.com/bytesparadise/libasciidoc/pkg/renderer/html5.Render(0xc0001674a0, 0x7e5880, 0xc00009e008, 0xc0001d4120, 0x3, 0x3)
        /go/src/github.com/bytesparadise/libasciidoc/pkg/renderer/html5/renderer.go:16 +0x3f
github.com/bytesparadise/libasciidoc.convertToHTML(0x7e6f40, 0xc00009a010, 0xc0001d40f0, 0xc0001d4120, 0x3, 0x3, 0xc0001d4150, 0xc215f0, 0x0, 0x0, ...)
        /go/src/github.com/bytesparadise/libasciidoc/libasciidoc.go:62 +0x105
github.com/bytesparadise/libasciidoc.ConvertToHTML(0x7e6f40, 0xc00009a010, 0x7e5860, 0xc00015a220, 0x7e5880, 0xc00009e008, 0xc0001c1c90, 0x1, 0x1, 0xc000127bc8, ...)
        /go/src/github.com/bytesparadise/libasciidoc/libasciidoc.go:57 +0x3d0
github.com/bytesparadise/libasciidoc.ConvertFileToHTML(0x7e6f40, 0xc00009a010, 0x7fff8a228b03, 0xe, 0x7e5880, 0xc00009e008, 0xc000127c90, 0x1, 0x1, 0x0, ...)
        /go/src/github.com/bytesparadise/libasciidoc/libasciidoc.go:37 +0x116
main.NewRootCmd.func1(0xc000198000, 0xc000163180, 0x1, 0x4, 0x0, 0x0)
        /go/src/github.com/bytesparadise/libasciidoc/cmd/libasciidoc/root_cmd.go:37 +0x295
github.com/spf13/cobra.(*Command).execute(0xc000198000, 0xc0000a4010, 0x4, 0x4, 0xc000198000, 0xc0000a4010)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:762 +0x473
github.com/spf13/cobra.(*Command).ExecuteC(0xc000198000, 0x1, 0x0, 0xc000198280)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2fd
github.com/spf13/cobra.(*Command).Execute(0xc000198000, 0xc000127f70, 0x1)
        /go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:800 +0x2b
main.main()
        /go/src/github.com/bytesparadise/libasciidoc/cmd/libasciidoc/main.go:17 +0x97
```

--- a/pkg/renderer/html5/table_of_contents.go
+++ b/pkg/renderer/html5/table_of_contents.go
@@ -52,7 +52,7 @@ func renderTableOfContents(ctx *renderer.Context, m types.TableOfContentsMacro) 
 		return nil, errors.Wrapf(err, "error while rendering table of content")
 	}
 	if renderedSections == template.HTML("") {
-		// nothing to render (doc has no section)
+		// nothing to render (document has no section)
 		return []byte{}, nil
 	}
 	result := bytes.NewBuffer(nil)

--- a/pkg/renderer/html5/table_of_contents.go
+++ b/pkg/renderer/html5/table_of_contents.go
@@ -43,17 +43,21 @@ type TableOfContentsSection struct {
 	Level       int
 	Href        string
 	Title       template.HTML
-	Subelements *template.HTML
+	Subelements template.HTML
 }
 
 func renderTableOfContents(ctx *renderer.Context, m types.TableOfContentsMacro) ([]byte, error) {
-	result := bytes.NewBuffer(nil)
 	renderedSections, err := renderTableOfContentsSections(ctx, ctx.Document.Elements, 1)
 	if err != nil {
 		return nil, errors.Wrapf(err, "error while rendering table of content")
 	}
+	if renderedSections == template.HTML("") {
+		// nothing to render (doc has no section)
+		return []byte{}, nil
+	}
+	result := bytes.NewBuffer(nil)
 	err = tableOfContentTmpl.Execute(result, TableOfContents{
-		Content: *renderedSections,
+		Content: renderedSections,
 	})
 	if err != nil {
 		return nil, errors.Wrapf(err, "error while rendering table of content")
@@ -62,7 +66,7 @@ func renderTableOfContents(ctx *renderer.Context, m types.TableOfContentsMacro) 
 	return result.Bytes(), nil
 }
 
-func renderTableOfContentsSections(ctx *renderer.Context, elements []interface{}, currentLevel int) (*template.HTML, error) {
+func renderTableOfContentsSections(ctx *renderer.Context, elements []interface{}, currentLevel int) (template.HTML, error) {
 	sections := make([]TableOfContentsSection, 0)
 	for _, element := range elements {
 		log.Debugf("traversing document element of type %T", element)
@@ -70,17 +74,17 @@ func renderTableOfContentsSections(ctx *renderer.Context, elements []interface{}
 		case types.Section:
 			renderedTitle, err := renderElement(ctx, section.Title.Elements)
 			if err != nil {
-				return nil, errors.Wrapf(err, "error while rendering table of content section")
+				return template.HTML(""), errors.Wrapf(err, "error while rendering table of content section")
 			}
 			tocLevels, err := ctx.Document.Attributes.GetTOCLevels()
 			if err != nil {
-				return nil, errors.Wrapf(err, "error while rendering table of content section")
+				return template.HTML(""), errors.Wrapf(err, "error while rendering table of content section")
 			}
-			var renderedChildSections *template.HTML
+			var renderedChildSections template.HTML
 			if currentLevel < *tocLevels {
 				renderedChildSections, err = renderTableOfContentsSections(ctx, section.Elements, currentLevel+1)
 				if err != nil {
-					return nil, errors.Wrapf(err, "error while rendering table of content section")
+					return template.HTML(""), errors.Wrapf(err, "error while rendering table of content section")
 				}
 			}
 			var id string
@@ -97,7 +101,7 @@ func renderTableOfContentsSections(ctx *renderer.Context, elements []interface{}
 		}
 	}
 	if len(sections) == 0 {
-		return nil, nil
+		return template.HTML(""), nil
 	}
 	resultBuf := bytes.NewBuffer(nil)
 	err := tableOfContentSectionSetTmpl.Execute(resultBuf, TableOfContentsSectionGroup{
@@ -105,9 +109,8 @@ func renderTableOfContentsSections(ctx *renderer.Context, elements []interface{}
 		Elements: sections,
 	})
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to render document TOC")
+		return template.HTML(""), errors.Wrap(err, "failed to render document TOC")
 	}
 	log.Debugf("retrieved sections for TOC: %+v", sections)
-	result := template.HTML(resultBuf.String())
-	return &result, nil
+	return template.HTML(resultBuf.String()), nil
 }

--- a/pkg/renderer/html5/table_of_contents_test.go
+++ b/pkg/renderer/html5/table_of_contents_test.go
@@ -170,5 +170,18 @@ A preamble...
 			verify(GinkgoT(), expectedResult, actualContent)
 		})
 
+		It("document with no section", func() {
+			actualContent := `= sect0
+:toc:
+
+level 1 sections not exists.`
+
+			expectedResult := `<div class="paragraph">
+<p>level 1 sections not exists.</p>
+</div>`
+			verify(GinkgoT(), expectedResult, actualContent)
+
+		})
 	})
+
 })

